src/Agent/NewRelic/Agent/Core/Utilization/VendorInfo.cs

```diff
@@ -279,37 +282,51 @@ private string GetProcessEnvironmentVariable(string variableName)
 #if NETSTANDARD2_0
         public IVendorModel GetDockerVendorInfo(IFileReaderWrapper fileReaderWrapper)
         {
-                IVendorModel vendorModel = null;
+            IVendorModel vendorModel = null;
+            try
+            {
+                var fileContent = fileReaderWrapper.ReadAllText("/proc/self/mountinfo");
+                vendorModel = TryGetDockerCGroupV2(fileContent);
+                if (vendorModel == null)
+                    Log.Finest("Found /proc/self/mountinfo but failed to parse Docker container id.");
+
+            }
+            catch (Exception ex)
+            {
+                Log.Finest(ex, "Failed to parse Docker container id from /proc/self/mountinfo.");
+            }
+
+            if (vendorModel == null) // fall back to the v1 check if v2 wasn't successful
+            {
                 try
                 {
-                    var fileContent = fileReaderWrapper.ReadAllText("/proc/self/mountinfo");
-                    vendorModel = TryGetDockerCGroupV2(fileContent);
+                    var fileContent = fileReaderWrapper.ReadAllText("/proc/self/cgroup");
+                    vendorModel = TryGetDockerCGroupV1(fileContent);
                     if (vendorModel == null)
-                        Log.Finest("Found /proc/self/mountinfo but failed to parse Docker container id.");
-
+                        Log.Finest("Found /proc/self/cgroup but failed to parse Docker container id.");
                 }
                 catch (Exception ex)
                 {
-                    Log.Finest(ex, "Failed to parse Docker container id from /proc/self/mountinfo.");
+                    Log.Finest(ex, "Failed to parse Docker container id from /proc/self/cgroup.");
                 }
+            }
 
-                if (vendorModel == null) // fall back to the v1 check if v2 wasn't successful
+            if (vendorModel == null)
+            {
+                try
                 {
-                    try
-                    {
-                        var fileContent = fileReaderWrapper.ReadAllText("/proc/self/cgroup");
-                        vendorModel = TryGetDockerCGroupV1(fileContent);
-                        if (vendorModel == null)
-                            Log.Finest("Found /proc/self/cgroup but failed to parse Docker container id.");
-                    }
-                    catch (Exception ex)
-                    {
-                        Log.Finest(ex, "Failed to parse Docker container id from /proc/self/cgroup.");
-                        return null;
-                    }
+                    vendorModel = TryGetEcsFargateDockerId();
+                    if (vendorModel == null)
+                        Log.Finest("Found ECS_CONTAINER_METADATA_URI_V4 but failed to parse Docker container id.");
```

Comment:
Won't `vendorModel` be null in most cases? Such as if neither environment variable exists?

Reply:
Your right, I see the problem.  I'll get it fixed.